### ISSUE TITLE
null page creation with netlify CMS starter

### ISSUE THREAD

**Issue by aidalgol** [WhiteAbeLincoln/gatsby-transformer-remark-frontmatter#1]
I have tried using this plugin with netlify-templates/gatsby-starter-netlify-cms, and it causes Gatsby to create a "null" page, which fails.

1. Checkout https://github.com/aidalgol/gatsby-starter-netlify-cms/tree/transformer-remark
2. Run `npm install`
3. Run `npm run develop`
4. Observe error
```
 ERROR #11325 

Your site's "gatsby-node.js" created a page with a component that doesn't exist.

The path to the missing component is "/path/to/gatsby-starter-netlify-cms/src/templates/null.js"

The page object passed to createPage:
{
    "path": "/",
    "tags": null,
    "component": "/path/to/gatsby-starter-netlify-cms/src/templates/null.js",
    "context": {
        "id": "e4bc4da5-3e9a-564d-947c-ff26fe2e4c3c"
    }
}
```

**Comment by LincMitch** [comment]
Getting the same error.

**Comment by WhiteAbeLincoln** [comment]
Sorry guys. I've been a little busy with school. This hasn't been an issue with the website that I created the plugin for. I'll see if I can reproduce your problem this weekend.

**Comment by danielpeger** [comment]
hey! I solved this by adding a type filter to the graphql query in gatsby-node.js
```
const result = await graphql(`
  {
    allMarkdownRemark(
      filter: { parent: { internal: { type: { eq: "File" } } } }
    ) {
      edges {
        node {
          ...
```

**Comment by WhiteAbeLincoln** [comment]
Interesting. It might be that gatsby is seeing my FrontmatterMarkdownFile nodes as real File nodes and creating a page for them. This isn't happening on my website because I filter my automatically created pages to only those that have a `templateKey` field in the frontmatter, and @danielpeger solved this by explicitly filtering to only those nodes that have a real File parent.

Specifically, the MarkdownRemark nodes that this plugin creates will have null `fileAbsolutePath` fields, which I gatsby may use to create the page. I probably can somehow set this field to the path of the real File parent, but that will just cause other issues (probably duplicate pages will be created, or pages will not have the expected information).

Unfortunately, I don't think this is something we can fix. I mentioned it at the end of the [Readme](https://github.com/WhiteAbeLincoln/gatsby-transformer-remark-frontmatter#possible-issues), but often we do want to imitate the file node so that plugins like gatsby-remark-images still work.

I can add a warning to the readme to make sure to filter your automatically created pages, but I don't see any solution that can be added to my code. Feel free to send me other suggestions if you think of any!

**Comment by WhiteAbeLincoln** [comment]
TL;DR solution:

Add this type filter to the graphql query in your gatsby-node.js
```
allMarkdownRemark(filter: { parent: { internal: { type: { ne: "FrontmatterMarkdownFile" }}}}) {
...
}
```
